### PR TITLE
reference/configuration, faq: add both config and faq of server-version

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

+ add config of `server-version`.
+ add faq of `server-version`.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version, including v4.0 changes for now)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/2573
- Other reference link(s):

### CODE REVIEW

File: reference/configuration/tidb-server/configuration-file.md [hunk 1]
```diff
@@ -84,6 +84,14 @@ The TiDB configuration file supports more options than command-line parameters.
 - Default value: `false`
 - With this default setting, adding or removing the primary key constraint is not supported. You can enable this feature by setting `alter-primary-key` to `true`. However, if a table already exists before the switch is on, and the data type of its primary key column is an integer, dropping the primary key from the column is not possible even if you set this configuration item to `true`.
 
+### `server-version`
+
++ Modifies the version number returned by TiDB in the following situations:
```
Comment (reafans) [inline reference/configuration/tidb-server/configuration-file.md:89]:
```suggestion
+ Modifies the version string returned by TiDB in the following situations:
```

File: reference/configuration/tidb-server/configuration-file.md [hunk 2]
```diff
@@ -84,6 +84,14 @@ The TiDB configuration file supports more options than command-line parameters.
...
+    - When the built-in `VERSION()` function is used.
+    - When TiDB establishes the initial connection to the client and returns the initial handshake packet with version number of the server. For details, see [MySQL Initial Handshake Packet](https://dev.mysql.com/doc/internals/en/connection-phase-packets.html#packet-Protocol::Handshake).
```
Comment (reafans) [inline reference/configuration/tidb-server/configuration-file.md:91]:
```suggestion
    - When TiDB establishes the initial connection to the client and returns the initial handshake packet with version string of the server. For details, see [MySQL Initial Handshake Packet](https://dev.mysql.com/doc/internals/en/connection-phase-packets.html#packet-Protocol::Handshake).
```

File: faq/tidb.md [hunk 3]
```diff
@@ -117,6 +117,10 @@ The auto-increment ID feature in TiDB is only guaranteed to be automatically inc
 
 The configuration method of TiDB `sql_mode` is different from that of MySQL `sql_mode`. TiDB does not support using the configuration file to configure `sql\_mode` of the database; it only supports using the `set` command to configure `sql\_mode` of the database. You can use `set @@global.sql_mode = 'STRICT_TRANS_TABLES';` to configure it.
 
+#### Does TiDB support modifying the MySQL version number of the server to a specific one that is required by the security vulnerability scanning tool?
```
Comment (reafans) [inline faq/tidb.md:120]: How about use the version string instead of version number.

**Comment by reafans** [comment]
LGTM

File: reference/configuration/tidb-server/configuration-file.md [hunk 4]
```diff
@@ -84,6 +84,14 @@ The TiDB configuration file supports more options than command-line parameters.
 - Default value: `false`
 - With this default setting, adding or removing the primary key constraint is not supported. You can enable this feature by setting `alter-primary-key` to `true`. However, if a table already exists before the switch is on, and the data type of its primary key column is an integer, dropping the primary key from the column is not possible even if you set this configuration item to `true`.
 
+### `server-version`
+
++ Modifies the version string returned by TiDB in the following situations:
+    - When the built-in `VERSION()` function is used.
+    - When TiDB establishes the initial connection to the client and returns the initial handshake packet with version string of the server. For details, see [MySQL Initial Handshake Packet](https://dev.mysql.com/doc/internals/en/connection-phase-packets.html#packet-Protocol::Handshake).
++ Default value: ""
++ By default, the TiDB version is in the format of `5.7.${mysql_latest_minor_version}-TiDB-${tidb_version}`.
```
Comment (yikeke) [inline reference/configuration/tidb-server/configuration-file.md:93]:
```suggestion
+ By default, the format of the TiDB version string is `5.7.${mysql_latest_minor_version}-TiDB-${tidb_version}`.
```

**Pull request merged**